ciri/consensus/snapshot/local_snapshots/pruning_service.h

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2019 IOTA Stiftung
+ * https://github.com/iotaledger/entangled
+ *
+ * Refer to the LICENSE file for licensing information
+ */
+
+#ifndef __CONSENSUS_SNAPSHOT_LOCAL_SNAPSHOTS_PRUNING_SERVICE_H__
+#define __CONSENSUS_SNAPSHOT_LOCAL_SNAPSHOTS_PRUNING_SERVICE_H__
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#include "ciri/consensus/conf.h"
+#include "ciri/consensus/snapshot/snapshots_provider.h"
+#include "ciri/consensus/spent_addresses/spent_addresses_service.h"
+#include "ciri/consensus/tangle/tangle.h"
+#include "ciri/node/tips_cache.h"
+#include "utils/handles/cond.h"
+#include "utils/handles/thread.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef struct pruning_service_s {
+  bool running;
+  thread_handle_t pruning_service_thread;
+  cond_handle_t cond_pruning_service;
+  iota_consensus_conf_t const *conf;
+  uint64_t last_pruned_snapshot_index;
+  uint64_t last_snapshot_index_to_prune;
+  lock_handle_t lock_handle;
+  snapshot_t *new_snapshot;
+  spent_addresses_service_t *spent_addresses_service;
+  tips_cache_t *tips_cache;
+} pruning_service_t;
+
+/**
+ * Initializes a pruning service
+ *
+ * @param ps[out] The pruning service
```

Comment:
The `[]` need to be tied to the `@param` to be detected by Doxygen. Like `@param[out]`.